### DISCUSSION TITLE
Unable to prevent multiplayer voice chat's ear-grabbing animation

### DISCUSSION THREAD

**Discussion by luke-de-vos** [Alf21/TTT2#1664] [Q&A] [answered]
In a TTT2 server, I run [this emote addon](https://steamcommunity.com/sharedfiles/filedetails/?id=1655753632). The emotes are immediately interrupted/canceled if the player begins voice chat during it. 

To fix this, I want to prevent the voicechat animation. The wiki suggests simply [overriding this hook](https://wiki.facepunch.com/gmod/GM:GrabEarAnimation). I ran the following code in shared context at `addons/addon_name/lua/autorun/addon_name.lua`. 

```
-- override  GM:GrabEarAnimation( Player ply ) to prevent grabbing ear
hook.Add("GrabEarAnimation", "ForcePreventGrabEarAnimation", function()
    print("ear grabbing should not happen")
    return true
end)
```

This had no visible impact. The print statement didn't execute either. Perhaps TTT2 is doing something different to run that animation?

Does anyone know how I can prevent this animation, or interruption to a separate animation in general? 

Thank you.

**Comment by Histalek** [comment, marked as answer]
Overriding the `GrabEarAnimation` hook won't work because we explicitly perform the animation in the `PlayerStartVoice` hook [1].
This is done the same way in vanilla TTT [2].

Restructuring the TTT2 code and running the `GrabEarAnimation` hook instead might be an option to explore


Aside: You just made me aware of a possible bug :D

[1] https://github.com/TTT-2/TTT2/blob/d3801d6f9f9cb4ac70c13ec45b055c01ad311b5d/gamemodes/terrortown/gamemode/client/cl_voice.lua#L391
[2] https://github.com/Facepunch/garrysmod/blob/68366f07a2ce93389cb54a231e3bcee1a2a8509e/garrysmod/gamemodes/terrortown/gamemode/cl_voice.lua#L503

**Reply by luke-de-vos** [reply]
Got it, thanks for the links, I see now. And glad to be of service! Lol.

Editing TTT2 directly feels out of my depth at the moment, but following up on your links I discovered ttt_show_gestures. I've found found that I can disable that for the duration of the dance, that seems to solve the problem!
